### ISSUE TITLE
Frequent DB connection timeout

### ISSUE THREAD

**Issue by yomexzo** [balderdashy/sails-mysql#175]
I have been having issues with my sails app deployment. I often get the following stack trace saying sails-mysql / waterline experiences a timeout while trying to spawn a new connection. But in every case, the DB is always up and running as I can connect to it using mysql clients. It happens often but haven't been able to determine it's pattern or what triggers it.

	Error: connect ETIMEDOUT
	    at afterwards (/sites/smser/node_modules/sails-mysql/lib/connections/spawn.js:72:13)
	    at /sites/smser/node_modules/sails-mysql/lib/connections/spawn.js:40:7
	    at Pool.<anonymous> (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/Pool.js:47:16)
	    at Handshake.Sequence.end (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/protocol/sequences/Sequence.js:78:24)
	    at Protocol.handleNetworkError (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:282:14)
	    at PoolConnection.Connection._handleNetworkError (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/Connection.js:303:18)
	    at PoolConnection.Connection._handleConnectTimeout (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/Connection.js:299:8)
	    at Socket.g (events.js:180:16)
	    at Socket.emit (events.js:92:17)
	    at Socket._onTimeout (net.js:327:8)
	Details:  Error: Could not connect to MySQL:
	Error: connect ETIMEDOUT

**Comment by Globegitter** [comment]
I am experiencing the same occasionally using a docker-container. Has been hard for me so far figuring out if the mysql-adapter is the issue or if it is docker.
But I have seen https://github.com/balderdashy/sails-mysql/blob/3d05499b5416f36e262ffc201fc8f6fd2488399d/lib/connections/spawn.js#L28 and there now is a `connectTimeout` option: https://github.com/felixge/node-mysql#connection-options

It seems default connections are set here: https://github.com/balderdashy/sails-mysql/blob/3d9face8ccb13fa857c4c63ff2356afded5b280f/lib/adapter.js#L67 and as far as I can tell the config at runtime is loaded here: https://github.com/balderdashy/sails-mysql/blob/0b659dd4634803eadcce886b475d59b861f77c16/lib/connections/register.js#L37

But not entirely sure, just started to look into it, certainly worth adding an option to increase the timeout, maybe even aligning it with the `_hookTimeout`

**Comment by Peter-Barrett** [comment]
Out of interest what method are you using for your migrations? 'safe', 'alter' or 'drop'?

**Comment by luiselizondo** [comment]
I'm getting the same message when importing a 2000+ records each with 243 columns. I'm using safe.

**Comment by sailsbot** [comment]
Thanks for posting, @yomexzo.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
